common/src/Model/CollectMatchingNodesVisitor.h

```diff
@@ -101,14 +102,16 @@ namespace TrenchBroom {
 
         template <typename V, typename I>
         std::vector<Node*> collectMatchingNodes(I cur, I end, Node* root) {
-            std::vector<Node*> result;
+            kdl::vector_set<Node*> result;
             while (cur != end) {
                 V visitor(*cur);
                 root->acceptAndRecurse(visitor);
-                result = VectorUtils::setUnion(result, visitor.nodes());
+
+                const auto& nodes = visitor.nodes();
+                result.insert(std::begin(nodes), std::end(nodes));
                 ++cur;
             }
-            return result;
+            return result.release_data();
```

Comment:
what's this?

Reply:
It returns the underlying vector of the vector_set by moving, which is safe here since its not reused.